### PR TITLE
unify select with empty set

### PR DESCRIPTION
**Pull request by traiansf**
###### Reviewer checklist

- [ ] Test coverage: `stack test --coverage`
- [ ] Public API documentation: `stack haddock`
- [ ] Style conformance: `stylish-haskell`

---

### CODE REVIEW

File: kore/test/Test/Kore/Builtin/Set.hs [hunk 1]
```diff
@@ -313,6 +313,28 @@ test_unifyFramingVariable =
             (===) expect =<< evaluate (mkAnd patConcreteSet patFramedSet)
         )
 
+selectPatternGen :: Monad m => PropertyT m (CommonStepPattern Object)
```
Comment (traiansf) [inline kore/test/Test/Kore/Builtin/Set.hs]: Concat (Elem(X), frame:Set)

File: kore/test/Test/Kore/Builtin/Set.hs [hunk 2]
```diff
@@ -313,6 +313,67 @@ test_unifyFramingVariable =
...
+            let concreteSet = Set.empty
+            patFramedSet <- selectPatternGen id
+            revPatFramedSet <- selectPatternGen reverse
+            fnPatFramedSet <- selectFunctionPatternGen id
+            revFnPatFramedSet <- selectFunctionPatternGen reverse
+            let
+                patConcreteSet = asPattern concreteSet
+                expect = ExpandedPattern.bottom
+            --                          Set.empty /\ SetItem(X:Int) Rest:Set
+            (===) expect =<< evaluate (mkAnd patConcreteSet patFramedSet)
```
Comment (ttuegel) [inline kore/test/Test/Kore/Builtin/Set.hs]: You might give this a name, such as
```.hs
let doesNotUnify pat1 pat2 = (===) expect =<< evaluate (mkAnd pat1 pat2)
```

File: kore/test/Test/Kore/Builtin/Set.hs [hunk 3]
```diff
@@ -313,6 +313,67 @@ test_unifyFramingVariable =
...
+-- `Rest:Set SetItem(absInt(X:Int))`, respectively.
+selectFunctionPatternGen
+    :: Monad m
+    => ([CommonStepPattern Object] -> [CommonStepPattern Object])
+    -> PropertyT m (CommonStepPattern Object)
+selectFunctionPatternGen permutation = do
+    elementVar <- forAll (standaloneGen $ variableGen intSort)
+    frameVar <- forAll (standaloneGen $ variableGen setSort)
+    Monad.when (variableName elementVar == variableName frameVar) discard
+    let element = mkApp intSort absIntSymbol  [mkVar elementVar]
```
Comment (ttuegel) [inline kore/test/Test/Kore/Builtin/Set.hs]: You could reduce duplication between `selectPatternGen` and `selectFunctionPatternGen` by taking an argument which may (or may not) apply a symbol to the generated variable.

File: kore/test/Test/Kore/Builtin/Set.hs (hunk 2 above)
Comment (traiansf) [reply kore/test/Test/Kore/Builtin/Set.hs]: and then I can name patConcreteSet patEmptySet

**Pull request merged**